diesel/src/query_dsl/mod.rs

```diff
@@ -603,6 +618,12 @@ pub trait QueryDsl: Sized {
     ///
     /// If there is already a `WHERE` clause, the result will be `old AND new`.
     ///
+    /// Note that in order to use this method with [`.select`], you will need to use it before calling
```

Comment:
The important point here is not that `.filter` is called before or after `.select`, but that `.filter` needs to be called after `inner_join()`/`left_join()` if the where clause contains a column from the joined table.

Reply:
I think I have a problem understanding the `FILTER`  keyword in SQL. I need to check/learn that :slightly_smiling_face:

Reply:
It's just a "where", there's no filter keyword in SQL. Where being a reserved keyword in Rust, we renamed it filter in diesel.

Reply:
Oh yes, that's right, I forgot, thank you. Okay, I get it now, I will do it.